sycl/doc/extensions/experimental/sycl_ext_oneapi_memory_export.asciidoc

```diff
@@ -0,0 +1,338 @@
+memory resource handles that can be exported by this extension. The `opaque_fd`
+and `win32_nt` values are used during allocation of exportable memory to
+indicate the type of handle that will later be returned by the
+`export_device_mem_handle` function.
+
+The `opaque_fd` handle type corresponds to a POSIX file descriptor, which is
+represented by an `int`.
+
+The `win32_nt` handle type corresponds to a Windows NT handle, which is
+represented by a `void *`.
+
+The `exported_mem` template struct is specialized for each of the
+`export_external_mem_handle_type` values. Each specialization defines a `type`
+alias that represents the type of the exported memory handle for that specific
+handle.
+
+The `exported_mem_t` type alias allows the user to easily obtain the type of
+the exported memory handle based on the `export_external_mem_handle_type`
+template parameter.
+
+=== API of the extension
+
+```c++
+
+namespace sycl::ext::oneapi::experimental {
+
+void *alloc_exportable_device_mem(
+    size_t alignment, size_t size,
+    export_external_mem_handle_type exportMemHandleType,
+    const sycl::device &syclDevice, const sycl::context &syclContext,
+    const property_list& propList = {});
+
+void *alloc_exportable_device_mem(
+    size_t alignment, size_t size,
+    export_external_mem_handle_type exportMemHandleType,
+    const sycl::queue &syclQueue,
+    const property_list& propList = {});
+}
+```
+
+The `alloc_exportable_device_mem` function allocates memory on the device marked
+as having the ability to later export that memory to an external memory resource
+handle.
+
+Memory allocated through this function must only be freed using
+`free_exportable_mem`. Using `sycl::free` to deallocate memory allocated with
+this function results in undefined behavior.
+
+With the exception of the `sycl::free` function from the core SYCL
+specification, pointers to memory allocated through this function may be passed
+to any core SYCL specification API accepting device USM memory pointers.
+
+Memory allocated through this function is only available on device.
+
+Memory allocated through this function has a linear memory layout on the device 
+(which is the same as memory allocated by other USM allocation functions like 
+`sycl::malloc_device`).
+
+Zero or more properties can be passed in the `propList` parameter via an
+instance of `sycl::property_list`. Currently, this extension does not define
+any properties that can be used with this function, so the `propList` parameter
+is ignored and reserved for future use.
+
+If an invalid `exportMemHandleType` is passed, the function will throw a
+`sycl::exception` with the `errc::invalid` code.
+
+The passed `exportMemHandleType` must be supported by the operating system
+running the SYCL application. Passing an `exportMemHandleType` that is not
+supported by the operating system running the SYCL application results in
+undefined behavior.
```

Comment:
This should provide more information about which types are supported on each operating system.  I assume `win32_nt` is supported only on Windows, but it doesn't hurt to say this in the spec.  I don't actually know about `opaque_fd`.  Is this supported only on Linux, or is it also supported on Windows?